### ISSUE TITLE
Wrong buffer name for `gnus-article-buffer`

### ISSUE THREAD

**Issue by thierryvolpiatto** [djcb/mu#2139]
**Describe the bug**

While looking at bug #2132 and writing PR #2137, I noticed that the value of `gnus-article-buffer` may change if I use gnus along mu4e for e.g. looking at News with Gmane.
After viewing Gmane news with gnus the `gnus-article-buffer` value become something like "*Article Gmane emacs-devel*", this because gnus modify original value in `gnus-article-setup-buffer`.

**To Reproduce**
1) Open News in gnus
2) Open any mail with mu4e
See the value of  `gnus-article-buffer` is no more "*Article*".

**Environment**

OS: LinuxMint

**Checklist**
- [X] you are running vanilla emacs (i.e. without Doom, Evil, Spacemacs etc.) (otherwise, please try to reproduce without those
- [X] you are running mu4e without any third-party extensions (otherwise, please try to reproduce without those)
- [X] you are running either the latest 1.6.x release, or a 1.7.x development release (otherwise, please upgrade).

**Comment by thierryvolpiatto** [comment]
To fix it we can set it back to original value in `mu4e-view`, I can make a PR on top of #2137 once it will be merged:

```diff
diff --git a/mu4e/mu4e-view.el b/mu4e/mu4e-view.el
index 7227645e..25515b77 100644
--- a/mu4e/mu4e-view.el
+++ b/mu4e/mu4e-view.el
@@ -622,6 +622,7 @@ marking if it still had that."
 
     (when (bufferp gnus-article-buffer)
     (kill-buffer gnus-article-buffer))
+    (setq gnus-article-buffer mu4e-view-buffer-name)
   (with-current-buffer (get-buffer-create gnus-article-buffer)
     (let ((inhibit-read-only t))
       (remove-overlays (point-min)(point-max) 'mu4e-overlay t)

```

**Comment by djcb** [comment]
IIRC there were (are?) some parts in the Gnus code that expected "*Article*", so simply renaming it led to errors.

**Comment by thierryvolpiatto** [comment]
"Dirk-Jan C. Binnema" ***@***.***> writes:

> IIRC there were (are?) some parts in the Gnus code that expected "Article", so simply renaming it led to errors.

Do you mean "Article" or "*Article*" the default value is the later and
after grepping gnus code I see no place where it can led to errors
e.g. string= etc... And because "*Article*" is anyway the default value
when Gnus start I don't see where it could clash.
I am running actually resetting it in mu4e-view and using also gnus to
view News, I will let you know if something goes wrong, for now no errors.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android. *


-- 
Thierry

**Comment by thierryvolpiatto** [comment]
Read `"Article"` or `"*Article*"`.

**Comment by Chris00** [comment]
It seems that  `*Article*` is not cast in stone: the [name is different for newsgroups](https://github.com/emacs-mirror/emacs/blob/2cd2fc74e474c158a1b0e6dfa29b49341262fcd9/lisp/gnus/gnus-art.el#L4514-L4520) but the default value of `gnus-article-buffer` [is `*Article*`](https://github.com/emacs-mirror/emacs/blob/2cd2fc74e474c158a1b0e6dfa29b49341262fcd9/lisp/gnus/gnus.el#L2428).  The only place `gnus-article-buffer` is compared to `*Article*` is to [conditionally execute `gnus-article-setup-buffer`](https://github.com/emacs-mirror/emacs/blob/2cd2fc74e474c158a1b0e6dfa29b49341262fcd9/lisp/gnus/gnus-sum.el#L5858-L5860).

**Comment by thierryvolpiatto** [comment]
Yes, right, more exactly in `gnus-select-newsgroup`, here is the code:

```elisp
      ;; Set up the article buffer now, if necessary.
      (unless (and gnus-single-article-buffer
		   (equal gnus-article-buffer "*Article*"))
	(gnus-article-setup-buffer))
```
So as you can see no problem when `gnus-article-buffer` is set with its default value, Gnus reset it to its needs, so I see no problems if we reset it to `*Article*` in mu4e.
